ta/pkcs11/src/processing.c

```diff
@@ -406,13 +420,59 @@ enum pkcs11_rc entry_processing_step(struct pkcs11_client *client,
 	if (rc)
 		return rc;
 
+	if (function == PKCS11_FUNCTION_DIGEST) {
+		if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
```

Comment:
maybe `switch` instead of `if`? Would look cleaner to me.

Reply:
I am OK to use switch in these two cases if this is what is wanted -- we didn't have definite decision on last discussion when to go for switch and when not ;).

In these cases it may even be clearer with if's

But I'll make [review] commits so we can see which feels better.

Reply:
I think we can remove an indent level:
```c
	if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
		assert(function == PKCS11_FUNCTION_DIGEST);

		rc = serialargs_get(&ctrlargs, &key_handle, sizeof(uint32_t));
		if (rc)
			return rc;
	}
```

Reply:
OK